### ISSUE TITLE
pyi-error caused by circular dependency

### ISSUE THREAD

**Issue by phamstered** [google/pytype#355]
I have a project which in the following structure, 

```
testpkg/
├── __init__.py
└── pkg
    ├── A.py
    ├── B.py
    └── __init__.py
```

File `testpkg/pkg/__init__.py`

```

class Common(object):

    __slots__ = ()

    def myprint(self):
        print(self.__class__)


from .A import A
from .B import B


__all__ = [A, B]
```

File testpkg/pkg/A.py

```
from testpkg.pkg import Common

class A(Common):
    pass
```

File testpkg/pkg/B.py

```
from testpkg.pkg import Common

class B(Common):
    pass
```

When I run `pytype testpkg -v 2` command, it rasied 
```
INFO No config file specified, and no [pytype] section in setup.cfg. Using default configuration.
INFO Configuration:
  exclude = set()
  inputs = {'/home/lyco/ConfigDispatcher/testpkg/main.py', '/home/lyco/ConfigDispatcher/testpkg/pkg/B.py', '/home/lyco/ConfigDispatcher/testpkg/__init__.py', '/home/lyco/ConfigDispatcher/testpkg/pkg/A.py', '/home/lyco/ConfigDispatcher/testpkg/pkg/__init__.py'}
  keep_going = False
  output = '/home/lyco/ConfigDispatcher/.pytype'
  pythonpath = ['/home/lyco/ConfigDispatcher']
  python_version = '3.6'
  disable = []
  report_errors = True
  protocols = False
  strict_import = False
Computing dependencies
INFO Source tree:
source: /home/lyco/ConfigDispatcher/testpkg/__init__.py
source: /home/lyco/ConfigDispatcher/testpkg/main.py
deps:
  [/home/lyco/ConfigDispatcher/testpkg/pkg/A.py->/home/lyco/ConfigDispatcher/testpkg/pkg/B.py->/home/lyco/ConfigDispatcher/testpkg/pkg/__init__.py]
source: [/home/lyco/ConfigDispatcher/testpkg/pkg/A.py->/home/lyco/ConfigDispatcher/testpkg/pkg/B.py->/home/lyco/ConfigDispatcher/testpkg/pkg/__init__.py]
INFO ------------- Starting pytype run. -------------
INFO infer command: pytype-single --imports_info $imports --module-name $module -V 3.6 -o $out --no-report-errors --nofail --quick $in
INFO check command: pytype-single --imports_info $imports --module-name $module -V 3.6 -o $out --analyze-annotated --nofail --quick $in
INFO infer testpkg.pkg.A
  imports: /home/lyco/ConfigDispatcher/.pytype/imports/testpkg.pkg.A.imports-1
  deps: ()
  output: /home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/A.pyi-1
INFO infer testpkg.pkg.B
  imports: /home/lyco/ConfigDispatcher/.pytype/imports/testpkg.pkg.B.imports-1
  deps: ()
  output: /home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/B.pyi-1
INFO infer testpkg.pkg.__init__
  imports: /home/lyco/ConfigDispatcher/.pytype/imports/testpkg.pkg.__init__.imports-1
  deps: ()
  output: /home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/__init__.pyi-1
INFO check testpkg.pkg.A
  imports: /home/lyco/ConfigDispatcher/.pytype/imports/testpkg.pkg.A.imports
  deps: ('/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/A.pyi-1', '/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/B.pyi-1', '/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/__init__.pyi-1')
  output: /home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/A.pyi
INFO check testpkg.pkg.B
  imports: /home/lyco/ConfigDispatcher/.pytype/imports/testpkg.pkg.B.imports
  deps: ('/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/A.pyi', '/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/B.pyi-1', '/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/__init__.pyi-1')
  output: /home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/B.pyi
INFO check testpkg.pkg.__init__
  imports: /home/lyco/ConfigDispatcher/.pytype/imports/testpkg.pkg.__init__.imports
  deps: ('/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/A.pyi', '/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/B.pyi', '/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/__init__.pyi-1')
  output: /home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/__init__.pyi
INFO check testpkg.main
  imports: /home/lyco/ConfigDispatcher/.pytype/imports/testpkg.main.imports
  deps: ('/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/A.pyi', '/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/B.pyi', '/home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/__init__.pyi')
  output: /home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/main.pyi
INFO check testpkg.__init__
  imports: /home/lyco/ConfigDispatcher/.pytype/imports/testpkg.__init__.imports
  deps: ()
  output: /home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/__init__.pyi
Analyzing 5 sources with 0 local dependencies
ninja: Entering directory `/home/lyco/ConfigDispatcher/.pytype'
[1/2] pytype-single --imports_info /home/lyco/ConfigDispatcher/.pytype/imports/testpkg.pkg.__init__.imports --module-name testpkg.pkg.__init__ -V 3.6 -o /home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/__init__.pyi --analyze-annotated --nofail --quick /home/lyco/ConfigDispatcher/testpkg/pkg/__init__.py
FAILED: /home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/__init__.pyi 
pytype-single --imports_info /home/lyco/ConfigDispatcher/.pytype/imports/testpkg.pkg.__init__.imports --module-name testpkg.pkg.__init__ -V 3.6 -o /home/lyco/ConfigDispatcher/.pytype/pyi/testpkg/pkg/__init__.pyi --analyze-annotated --nofail --quick /home/lyco/ConfigDispatcher/testpkg/pkg/__init__.py
File "/home/lyco/ConfigDispatcher/testpkg/pkg/__init__.py", line 11, in <module>: Couldn't import pyi for 'A' [pyi-error]
  No Common in module testpkg.pkg, referenced from 'testpkg.pkg.A'
File "/home/lyco/ConfigDispatcher/testpkg/pkg/__init__.py", line 12, in <module>: Couldn't import pyi for 'B' [pyi-error]
  No Common in module testpkg.pkg, referenced from 'testpkg.pkg.B'

For more details, see https://google.github.io/pytype/errors.html#pyi-error.
ninja: build stopped: subcommand failed.

```

**Comment by rchen152** [comment]
Thanks for the report! (and for providing such a clear and complete repro of the bug) This looks similar to https://github.com/google/pytype/issues/320, in that pytype is being tripped up by what it sees as a circular dependency (`testpkg.pkg` imports `testpkg.pkg.A` imports `testpkg.pkg`), but in this case, I looked at the generated pyi files and pytype did run over the cycle twice as expected. So this is probably a bug in how load_pytd.py looks up pyi files rather than in how the files are generated.

**Comment by pavelpatrin** [comment]
Also have same problem.

**Comment by rchen152** [comment]
This looks pretty similar to #760, but the proposed fix for that one doesn't fix this issue, so there must be something else going on.

**Comment by Cebtenzzre** [comment]
The committed fix for #760, 4efe3c1560c47c47157cad7600a1efc0e70e9eb9, actually does fix this issue.

**Comment by rchen152** [comment]
Good to know, thanks for checking.
